src/dataflow-types/src/types.rs

```diff
@@ -2758,6 +2758,24 @@ pub mod sources {
             Self::try_from(proto).map_err(|err| err.to_string())
         }
     }
+
+    #[cfg(test)]
+    mod tests {
+        use super::*;
+        use mz_repr::proto::protobuf_roundtrip;
+        use proptest::prelude::*;
+
+        proptest! {
+            #![proptest_config(ProptestConfig::with_cases(32))]
+
+            #[test]
+            fn external_source_connector_protobuf_roundtrip(expect in any::<ExternalSourceConnector>()) {
+                let actual = protobuf_roundtrip::<_, ProtoExternalSourceConnector>(&expect);
+                assert!(actual.is_ok());
+                assert_eq!(actual.unwrap(), expect);
+            }
+        }
+    }
```

Comment:
is this guaranteed to test EVERY enum variant, recursively? Thats the kind of test I would want, and it taking 6-8s seems totally reasonable to me

Reply:
I'm not sure! My understanding of the way `proptest` generates samples for `enum`s is that yes, it should cover all cases (it was at least very consistent in finding bugs I had while developing this!), but `proptest` as a whole is still a bit of magic that I don't fully get right now.

Reply:
Ill ask in slack!